[Kaka0 Adult] 결제 기능 구현

hyo-jin-jeong opened the pull request:
>wecode PR message template 및 체크 리스트 입니다. 
아래 사항들을 전부 작성/체크 하시고 PR 해주세요!

## 수정 사항 간략한 한줄 요약
(여기에 수정 사항에 대한 간략한 한줄 요약/제목 작성해 주세요.)
- 결제 기능 구현
		 
## 수정 사항들 자세한 내용
(여기에 수정 사항에 대한 자세한 내용을 작성해 주세요.)
- OrderView 생성
- post method 생성
- transaction 처리 

## 기타 질문 및 특이 사항
(궁금한 사항들, 하면서 느낀 점들 공유해주실 것이 있으면 자유롭게 작성해 주세요.)
- 찾아보니 @transaction.atomic 과 with transaction.atomic 둘 다 exception이 발생하면 rollback이 된다고 하는데 왜 @transaction.atomic으로 하면 keyerror 발생 시 rollback 되지 않는 것인지 궁금합니다. 혹시 로그인 데코레이터와 같이 쓰는것이 문제일까요?

## 체크 리스트 (아래 사항들이 전부 체크되어야만 merge가 됩니다!)
- [x] 필요한 test들을 완료하였고 기능이 제대로 실행되는지 확인 하였습니다.
- [x] Wecode의 코드 스타일 가이드에 맞추어 코드를 작성 하였습니다.
- [x] 제가 의도한 파일들과 수정 사항들만 커밋이 된 것을 확인 하였습니다.
- [x] 본 수정 사항들을 팀원들과 사전에 상의하였고 팀원들 모두 해당 PR에 대하여 알고 있습니다.

hyo-jin-jeong committed:
Create: OrderView post method 구현

hyo-jin-jeong committed:
Modify: except 처리

hyo-jin-jeong committed:
Modify: OrderView post method 수정

hyo-jin-jeong committed:
Modify: get method json id 값 추가

Kang-kyunghun commented on orders/views.py:
```diff
@@ -62,16 +64,56 @@ def delete(self, request, order_item):
                 'image_url'    : order_item.product.imageurl_set.order_by('id')[0].url,
                 'selected'     : order_item.selected
-            }for order_item in OrderItem.objects.filter(
+            }for index, order_item in enumerate(OrderItem.objects.filter(
```
여기서 enumerate를 사용한 이유가 무엇인지 코멘트로 남겨 주새요.

Kang-kyunghun commented on orders/views.py:
```diff
@@ -62,16 +64,56 @@ def delete(self, request, order_item):
+                    address      = recipient_info['address'],
+                    name         = recipient_info['name'],
+                    phone_number = recipient_info['phone_number'],
+                    request      = request
```
```suggestion
                    request      = request if request else ""
```

hyo-jin-jeong committed:
Modify: merge 후 충돌 해결 & 멘토님 리뷰 반영
- order_item_count 변수 생성
- RecipientInfo model request 변수 -> comment 변경
- order - recipient 연결

hyo-jin-jeong committed:
Refactor: 공백 제거

Kang-kyunghun commented:
고생 많으셨습니다~~!! 👍
